active/XamlServiceProvider/XamlServiceProviderApi.md

```diff
@@ -0,0 +1,272 @@
+* `IRootObjectProvider`, which provides a reference to the root object in the markup. This aligns with WPF's [IRootObjectProvider](https://docs.microsoft.com/dotnet/api/System.Xaml.IRootObjectProvider).
+* `IUriContext` , which provides the base URI of the markup. This aligns with WPF's [IUriContext](https://docs.microsoft.com/dotnet/api/System.Windows.Markup.IUriContext).
+* `IXamlTypeResolver`, which binds a Xaml markup name to a type. This aligns with WPF's [IXamlTypeResolver](https://docs.microsoft.com/en-us/dotnet/api/system.windows.markup.ixamltyperesolver).
+
+There are two overloads of the virtual ProvideValue method, with/without the IXamlServiceProvider parameter. During Xaml markup loading, ProvideValue(IXamlServiceProvider) is called. Its virtual implementatiohn calls ProvideValue(). So not overriding ProvideValue(IXamlServiceProvider), or calling the base implementation of it, calls ProvideValue().
+
+## IXamlServiceProvider interface
+Gets the service object of the specified type.
+
+## IProvideValueTarget interface
+Provides a target object and property.
+
+Xaml markup extensions offer this interface in the IXamlServiceProvider parameter.  The target object/property are the instance and property identifier that the markup extension is being set on.
+
+## IRootObjectProvider interface
+Provides a root object
+
+Xaml markup extensions offer this interface in the IXamlServiceProvider parameter.  The root object is the root of the markup being loaded.
+
+## IUriContext interface
+Provides a URI
+
+Xaml markup extensions offer this interface in the IXamlServiceProvider parameter.  The URI is that of the markup file being loaded (the `resourceLocator` value passed to [Application.LoadComponent](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Application.LoadComponent)). The BaseUri property could be null.
+
+# API Details
+
+```cs
+namespace Microsoft.UI.Xaml
+{
+    interface IXamlServiceProvider 
+    {
+        Object GetService(Windows.UI.Xaml.Interop.TypeName type);
+    };
+}
+
+namespace Microsoft.UI.Xaml.Markup
+{
+    interface IProvideValueTarget 
+    {
+        Object TargetObject{ get; };
+        Object TargetProperty{ get; };
+    };
+
+    interface IRootObjectProvider 
+    {
+        Object RootObject{ get; };
+    };
+
+    interface IUriContext 
+    {
+        Windows.Foundation.Uri BaseUri{ get; };
+    };
+
+    interface IXamlTypeResolver 
+    {
+        Windows.UI.Xaml.Interop.TypeName Resolve(String qualifiedTypeName);
+    };
+
+    unsealed runtimeclass MarkupExtension
+    {
+        protected MarkupExtension();
+    
+        overridable Object ProvideValue();
+
+        // New
+        overridable Object ProvideValue(Windows.UI.Xaml.IXamlServiceProvider serviceProvider);
+    };
+
+    runtimeclass ProvideValueTargetProperty

```

Comment:
What other Targets are expected?

Reply:
Do you mean what other types are expected to be queried for?

Reply:
Oh, as in what targets other than ProvideValue. MarkupExtension exists for the ProvideValue call, with the output being set on a target. So it's really just about this one target.